### ISSUE TITLE
Ok to pass in View to RxPresenter?

### ISSUE THREAD

**Issue by sreejithraman** [konmik/nucleus#67]
As I understand `getView()` is being deprecated. However there are a couple instances where I would need a view without using Rx. In those cases, is it ok to pass in the view to an RxPresenter? I am using the Facebook SDK in my app. When I want to log in a user with Facebook, the SDK uses callbacks to allow me to handle `onSucess()`, `onCancel()`, and `onError()`. Going from the principals of MVP, all the login logic should go inside a Presenter. So in those cases can I do something like this: 

````Java
public void loginWithFacebook(final LogInMainFragment logInMainFragment) {
        LoginManager.getInstance().logOut();//make sure that user is logged out

        LoginManager.getInstance().registerCallback(logInMainFragment.getFacebookCallbackManager(),
                new FacebookCallback<LoginResult>() {
                    @Override
                    public void onSuccess(final LoginResult loginResult) {
                             // do things with the view (like starting a new activity) which requires me to have an instance of the view
                    }

                    @Override
                    public void onCancel() {
                        //do things like hiding progress bars, or displaying dialogs
                    }

                    @Override
                    public void onError(FacebookException e) {
                         //do things like hiding progress bars, or displaying dialogs
                    }
                });

        LoginManager.getInstance().logInWithReadPermissions(logInMainFragment, FacebookUtils.getFacebookLoginPermissionsList()); 

    }
````

Basically in each of those methods, I need a reference to the view to update the view. How would I be able to get references to the view in those situations. I know I am probably missing something, so could you please point me in the right direction.

**Comment by konmik** [comment]
Hi,

The logic should be in the place where it is more convenient for you. I think that one should not put everything into a presenter just for the sake of having something in the presenter.

I see four possible solutions: have your code inside of View, use `onTakeView`, use `view().filter(v -> v != null)`, build your own delivery method using raw RxJava operators.

**Comment by sreejithraman** [comment]
Correct me if I am wrong but the way `view()` works is, it returns a `BehaviorSubject<View>`. And so until `onTakeView()` is called, the  `BehaviorSubject<View>` will not emit anything (therefore I don't need to worry about the view being null).

**Comment by konmik** [comment]
Yes, but later you can get null because of onPause event.

**Comment by sreejithraman** [comment]
I guess that kinda pokes at the question I was having in my mind. What exactly is the life cycle of your presenters? Will a presenter every hit onPause when a view is attached? Is the lifecycle somehow tied to the view's lifecycle?

**Comment by konmik** [comment]
I advise to read the sources. They are very simple.

https://github.com/konmik/nucleus/blob/master/nucleus/src/main/java/nucleus/view/PresenterLifecycleDelegate.java#L103

https://github.com/konmik/nucleus/blob/master/nucleus/src/main/java/nucleus/view/NucleusActivity.java#L76

https://github.com/konmik/nucleus/blob/master/nucleus/src/main/java/nucleus/view/NucleusFragment.java#L74

https://github.com/konmik/nucleus/blob/master/nucleus/src/main/java/nucleus/view/NucleusLayout.java#L110
